### PR TITLE
feat: Release kedro-datasets version 3.0.0

### PR DESCRIPTION
**Pull request by lrcouto**
## Description
Release `kedro-datasets 3.0.0`

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

**Comment by merelcht** [comment]
I realised when the `ibis` dataset was merged that we have changed the release notes format for when a new dataset is added. We used to format it in a table like here: https://github.com/kedro-org/kedro-plugins/blob/38ad6196b5bb82406b5ecdac3b6d380ce5dfc394/kedro-datasets/RELEASE.md#major-features-and-improvements-11 

I would actually like to bring that format back, because I think it makes new datasets much more visible. Maybe we can do a bit of release notes cleanup for this major release?

File: kedro-datasets/RELEASE.md
Comment (merelcht) [inline kedro-datasets/RELEASE.md:1]: Yes, that's what I meant! Amazing, thanks @elena-khaustova-qb 😄

**Comment by noklam** [comment]
Let's wait  #646 for a bit.

File: kedro-datasets/RELEASE.md
Comment (noklam) [reply kedro-datasets/RELEASE.md:1]: Promoting the `/table` Github command here to create the table. You may need to enable beta feature of Github though.

File: kedro-datasets/RELEASE.md [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
+
+# Release 3.0.0
+## Major features and improvements
+
+* Added the following new datasets:
+
+| Type                   | Description                                               | Location                |
+|------------------------|-----------------------------------------------------------|-------------------------|
+| `netcdf.NetCDFDataset` | A dataset for loading and saving `*.nc` files.            | `kedro_datasets.netcdf` |
+| `ibis.NetCDFDataset`   | A dataset for loading and saving using Ibis's backends.   | `kedro_datasets.ibis`   |
```
Comment (ankatiyar) [inline kedro-datasets/RELEASE.md:16]: Is it not `ibis.TableDataset`?

Comment (ElenaKhaustova) [reply kedro-datasets/RELEASE.md]: Fixed  🙈

**Review by merelcht — APPROVED** [review]
🕺

**Review by ankatiyar — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**